rpcserver.go

```diff
@@ -2092,8 +2092,18 @@ func (r *rpcServer) BurnAsset(ctx context.Context,
 
 	var groupKey *btcec.PublicKey
 	assetGroup, err := r.cfg.TapAddrBook.QueryAssetGroup(ctx, assetID)
-	if err == nil && assetGroup.GroupKey != nil {
+	switch {
+	case err == nil:
```

Comment:
I think we still want the `assetGroup.GroupKey != nil` safety check. Because looking at [this](https://github.com/lightninglabs/taproot-assets/blob/bc077ce8844fb1b2f0040979118193d01636c323/tapdb/addrs.go#L952), it can be that we return without an error but with the group being `nil`.